pwned_passwords_django/tests/test_api.py

```diff
@@ -77,8 +81,69 @@ def test_empty_response(self):
         with mock.patch('requests.get', request_mock):
             result = api.pwned_password(self.sample_password)
             request_mock.assert_called_with(
-                api.API_ENDPOINT.format(
+                url=api.API_ENDPOINT.format(
                     self.sample_password_prefix
-                )
+                ),
+                timeout=0.6,
+            )
+            self.assertEqual(None, result)
+
+    def test_bad_text(self):
+        """
+        Handle non-numeric count gracefully
+
+        """
+        request_mock = self._get_mock(
+            response_text='{}:xxx'.format(
+                self.sample_password_suffix
+            )
+        )
+        with mock.patch('requests.get', request_mock):
+            result = api.pwned_password(self.sample_password)
+            self.assertEqual(None, result)
+
+    def test_bad_response_no_colon(self):
+        """
+        Handle malformed response with no colon gracefully
+
+        """
+        request_mock = self._get_mock(
+            response_text=self.sample_password_suffix
+        )
+        with mock.patch('requests.get', request_mock):
+            result = api.pwned_password(self.sample_password)
+            self.assertEqual(None, result)
+
+    def test_bad_response_many_colons(self):
+        """
+        Handle malformed response with too many colons gracefully
+
+        """
+        request_mock = self._get_mock(
+            response_text='{}:123:xxx'.format(
+                self.sample_password_suffix
             )
+        )
+        with mock.patch('requests.get', request_mock):
+            result = api.pwned_password(self.sample_password)
+            self.assertEqual(None, result)
+
+    def test_timeout(self):
+        """
+        Connection timeout response is handled gracefully
+
+        """
+        request_mock = mock.MagicMock(side_effect=requests.ConnectTimeout())
```

Comment:
Probably `_get_mock()` on the base test case should learn how to handle this argument, by changing signature to `def _get_mock(self, response_text, **kwargs)` and then passing the kwargs when instantiating the `MagicMock`.